src/ngraph/pattern/matcher.cpp

```diff
@@ -38,6 +38,48 @@ namespace ngraph
                 begin(arguments), end(arguments)); //vector is needed for generating permutations
         }
 
+        bool Matcher::match_recurring_pattern(
+            std::shared_ptr<Node> graph,
+            std::shared_ptr<Node> pattern,
+            std::shared_ptr<op::Label> rpattern,
+            RPatternMap& patterns,
+            const std::set<std::shared_ptr<op::Label>>& correlated_patterns)
+        {
+            bool no_match = false;
+            Matcher m(pattern);
+            PatternMap previous_matches;
+
+            NGRAPH_DEBUG << "matching graph to " << graph->get_name() << std::endl;
+            while (m.match(graph, previous_matches))
+            {
+                no_match = true;
+                //for the next round
+                graph = m.m_pattern_map[rpattern];
+                NGRAPH_DEBUG << "setting graph to " << graph->get_name() << std::endl;
+                for (auto me : m.m_pattern_map)
+                {
```

Comment:
`auto&` ? I'm not sure what's in a pattern map. Is `me` something like a `map entry`? That would be a better variable name then.

Reply:
yes, that's pretty much what it stands for.  I can call it *local_match*
if matched a *cell* we go over all labels and add bound nodes to a global map.